optaplanner-persistence/optaplanner-persistence-xstream/src/main/java/org/optaplanner/persistence/xstream/impl/domain/solution/XStreamSolutionFileIO.java

```diff
@@ -61,11 +62,19 @@ public String getInputFileExtension() {
 
     @Override
     public Solution_ read(File inputSolutionFile) {
+        try(InputStream inputSolutionStream = Files.newInputStream(inputSolutionFile.toPath())) {
+            return read(inputSolutionStream);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Failed reading inputSolutionFile (" + inputSolutionFile + ").", e);
```

Comment:
This one is serious: if there an XStreamException or an IOException during `read(InputStream)`, the error message won't mention the inputSolutionFile, which is important as a breadcrumb (to figure out what 's wrong for the user).
Why? that read(InputStream) throws an IllegalSargumentException and that isn't caught.

xstream exceptions are very common btw. Either invalid xml or <localSearch> that has a <unexistingElement> (there is no XSD).

Thinking further:
if there is such an xml parsing or element nesting error for an resource (not file), then we still want the resource name.

Suggestion to fix all of this:
1) Don't add `read(InputStream()`. Add `readResource(String resource)` instead. That also decreases the boilerplate in the NQueensCosntraintProviderTest